src/ModelTable/ModelTableActionCell.tsx

```diff
@@ -34,7 +34,9 @@ export const ModelTableActionCell = ({
   };
   const onSaveHandler = handleSubmit(async (formData: DataType) => {
     const changedData = Object.fromEntries(
-      Object.entries(formData).filter((entry) => dirtyFields[entry[0]]),
+      Object.entries(formData)
+        .filter((entry) => dirtyFields[entry[0]])
+        .map((entry) => [entry[0], entry[1] === '' ? null : entry[1]]),
```

Comment:
You changing the value in Number input should have no affect on this since it uses `value` and not what is passed to the actual input.

Reply:
If you mean `value={value || ''}` in `NumberInput.tsx`, that is for the case if `value` given to it is null or undefined. When NumberInput is cleared and there is no number, the value of the input is `''`